File: 设计模式/程序/2-AbstractFactoryPattern/AbFactoryPattern(python)/AbstractFactory.py
from ShapeFactory import *
from ColorFactory import *

class FactoryProducer(object):
    def __init__(self):
        pass

    def GetFactory(self, factoryType):
        if factoryType == "Color":
            return ColorFactory()
        elif factoryType == "Shape":
            return ShapeFactory()
        else:
            print("No", factoryType, "factory")
            return None

File: 设计模式/程序/2-AbstractFactoryPattern/AbFactoryPattern(Cpp)/src/AbstractFactory.cpp
#include "AbstractFactory.h"
#include "ColorFactory.h"
#include "ShapeFactory.h"

#include <iostream>
#include <string>


namespace AbstratFactoryPattern
{
    /*********************************************************
     * @brief Get the Factory object
     * @param factoryType: thefactory you need to create
     * @return AbstractFactory* the object pointer
     *********************************************************/
    AbstractFactory* GetFactory(const std::string& factoryType)
    {
        if (factoryType == "COLOR")
            return new ColorFactory;
        else if (factoryType == "SHAPE")
            return new ShapeFactory;
        return nullptr;
    }
}